driver-core/src/main/com/mongodb/internal/operation/QueryBatchCursor.java

```diff
@@ -320,83 +322,311 @@ private BsonDocument asGetMoreCommandDocument() {
         return document;
     }
 
-    private void initFromQueryResult(final QueryResult<T> queryResult) {
-        serverCursor = queryResult.getCursor();
+    @Nullable
+    private ServerCursor initFromQueryResult(final QueryResult<T> queryResult) {
```

Comment:
The changes in `initFromQueryResult` and `initFromCommandResult` were made to move the code changing the reference to `ServerCursor` to `getMore`, such that it is more explicit and, what is more important, to reduce the number of methods that use releasable resources managed by `ResourceManager` either directly or indirectly.